kvbc/include/db_checkpoint.h

```diff
@@ -0,0 +1,142 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "status.hpp"
+#include "db_adapter_interface.h"
+#include "db_interfaces.h"
+#include "Serializable.h"
+#include "PersistentStorage.hpp"
+#include <optional>
+#include <string>
+#include <vector>
+#include <mutex>
+#include <queue>
+#include <condition_variable>
+#include <chrono>
+#include <bftengine/DbCheckpointMetadata.hpp>
+#include "Metrics.hpp"
+#include <algorithm>
+#if __has_include(<filesystem>)
+#include <filesystem>
+namespace fs = std::filesystem;
+#elif __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+namespace _fs = std::experimental::filesystem;
+#else
+#error "Missing filesystem support"
+#endif
+namespace concord::kvbc {
+using std::chrono::duration_cast;
+using CheckpointId = bftEngine::impl::CheckpointId;
+using DbCheckpointMetadata = bftEngine::impl::DbCheckpointMetadata;
+using Status = concordUtils::Status;
+using SystemClock = std::chrono::system_clock;
+using Seconds = std::chrono::seconds;
+class IDbCheckPointManager {
```

Comment:
As mentioned earlier, the name of this interface can be confused with `DBCheckPointManager` itself, 
and shouldn't be defined in `kvbc`